core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -148,12 +148,49 @@ public static List<String> statisticsFilesLocations(Table table) {
    * @param table table for which statistics files needs to be listed
    * @param predicate predicate for filtering the statistics files
    * @return the location of statistics files
+   * @deprecated use the {@code statisticsFilesLocationsForSnapshots(table, snapshotIds)} instead.
```

Comment:
Shall we add a note on when it was deprecated (1.5.0) and when it will be removed (1.6.0)?

Reply:
I have added this already in previous commit.